2025-08-11 02:40:01,775 INFO    MainThread:10987 [wandb_setup.py:_flush():80] Current SDK version is 0.21.1
2025-08-11 02:40:01,775 INFO    MainThread:10987 [wandb_setup.py:_flush():80] Configure stats pid to 10987
2025-08-11 02:40:01,775 INFO    MainThread:10987 [wandb_setup.py:_flush():80] Loading settings from /home/ubuntu/.config/wandb/settings
2025-08-11 02:40:01,775 INFO    MainThread:10987 [wandb_setup.py:_flush():80] Loading settings from /home/ubuntu/babylm-poincare-structformer/wandb/settings
2025-08-11 02:40:01,775 INFO    MainThread:10987 [wandb_setup.py:_flush():80] Loading settings from environment variables
2025-08-11 02:40:01,775 INFO    MainThread:10987 [wandb_init.py:setup_run_log_directory():703] Logging user logs to logs/structformer_run/wandb/run-20250811_024001-mm65fu44/logs/debug.log
2025-08-11 02:40:01,775 INFO    MainThread:10987 [wandb_init.py:setup_run_log_directory():704] Logging internal logs to logs/structformer_run/wandb/run-20250811_024001-mm65fu44/logs/debug-internal.log
2025-08-11 02:40:01,775 INFO    MainThread:10987 [wandb_init.py:init():830] calling init triggers
2025-08-11 02:40:01,775 INFO    MainThread:10987 [wandb_init.py:init():835] wandb.init called with sweep_config: {}
config: {'model_type': 'structformer_poincare', 'framework': 'jax_flax', 'vocab_size': 50257, 'hidden_dim': 256, 'num_layers': 8, 'num_heads': 8, 'max_length': 128, 'hyperbolic_c': 1.0, 'lr_ce': 0.0002, 'lr_riem': 0.0001, 'lambda_poincare': None, 'batch_size': 128, 'max_words': 100000000, '_wandb': {'code_path': 'code/scripts/train.py'}}
2025-08-11 02:40:01,775 INFO    MainThread:10987 [wandb_init.py:init():871] starting backend
2025-08-11 02:40:01,979 INFO    MainThread:10987 [wandb_init.py:init():874] sending inform_init request
2025-08-11 02:40:01,981 INFO    MainThread:10987 [wandb_init.py:init():882] backend started and connected
2025-08-11 02:40:01,982 INFO    MainThread:10987 [wandb_init.py:init():953] updated telemetry
2025-08-11 02:40:01,985 INFO    MainThread:10987 [wandb_init.py:init():977] communicating run to backend with 90.0 second timeout
2025-08-11 02:40:02,311 INFO    MainThread:10987 [wandb_init.py:init():1029] starting run threads in backend
2025-08-11 02:40:02,371 INFO    MainThread:10987 [wandb_run.py:_console_start():2494] atexit reg
2025-08-11 02:40:02,371 INFO    MainThread:10987 [wandb_run.py:_redirect():2342] redirect: wrap_raw
2025-08-11 02:40:02,371 INFO    MainThread:10987 [wandb_run.py:_redirect():2411] Wrapping output streams.
2025-08-11 02:40:02,371 INFO    MainThread:10987 [wandb_run.py:_redirect():2434] Redirects installed.
2025-08-11 02:40:02,372 INFO    MainThread:10987 [wandb_init.py:init():1075] run started, returning control to user process
2025-08-11 02:40:02,528 INFO    MsgRouterThr:10987 [mailbox.py:close():129] [no run ID] Closing mailbox, abandoning 1 handles.
2025-08-11 02:40:03,015 ERROR   MainThread:10987 [redirect.py:_on_write():664] [all runs] error in stderr callback
Traceback (most recent call last):
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/lib/redirect.py", line 662, in _on_write
    cb(written_data)
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 2421, in <lambda>
    lambda data: self._console_raw_callback("stderr", data),
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 399, in wrapper
    return func(self, *args, **kwargs)
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 457, in wrapper_fn
    return func(self, *args, **kwargs)
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 1583, in _console_raw_callback
    self._backend.interface.publish_output_raw(name, data)
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/interface/interface.py", line 771, in publish_output_raw
    self._publish_output_raw(o)
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/interface/interface_shared.py", line 38, in _publish_output_raw
    self._publish(rec)
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/interface/interface_sock.py", line 39, in _publish
    self._sock_client.send_record_publish(record)
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 166, in send_record_publish
    self.send_server_request(server_req)
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 146, in send_server_request
    self._send_message(msg)
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 143, in _send_message
    self._sendall_with_error_handle(header + data)
  File "/home/ubuntu/babylm-poincare-structformer/bb-env/lib/python3.10/site-packages/wandb/sdk/lib/sock_client.py", line 122, in _sendall_with_error_handle
    sent = self._sock.send(data)
BrokenPipeError: [Errno 32] Broken pipe
